File: sdks/java/src/main/java/dev/atap/sdk/oauth/OAuthToken.java
package dev.atap.sdk.oauth;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * An OAuth 2.1 token response.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class OAuthToken {

    @JsonProperty("access_token")
    private String accessToken;

    @JsonProperty("token_type")
    private String tokenType;

    @JsonProperty("expires_in")
    private int expiresIn;

    @JsonProperty("scope")
    private String scope;

    @JsonProperty("refresh_token")
    private String refreshToken;

    public OAuthToken() {
        this.tokenType = "DPoP";
        this.expiresIn = 3600;
    }

    public OAuthToken(String accessToken, String tokenType, int expiresIn, String scope, String refreshToken) {
        this.accessToken = accessToken;
        this.tokenType = tokenType;
        this.expiresIn = expiresIn;
        this.scope = scope;
        this.refreshToken = refreshToken;
    }

    public String getAccessToken() { return accessToken; }
    public void setAccessToken(String accessToken) { this.accessToken = accessToken; }

    public String getTokenType() { return tokenType; }
    public void setTokenType(String tokenType) { this.tokenType = tokenType; }

    public int getExpiresIn() { return expiresIn; }
    public void setExpiresIn(int expiresIn) { this.expiresIn = expiresIn; }

    public String getScope() { return scope; }
    public void setScope(String scope) { this.scope = scope; }

    public String getRefreshToken() { return refreshToken; }
    public void setRefreshToken(String refreshToken) { this.refreshToken = refreshToken; }
}

File: sdks/php/src/Model/OAuthToken.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Model;

/**
 * An OAuth 2.1 token response.
 */
final class OAuthToken
{
    public function __construct(
        public readonly string $accessToken,
        public readonly string $tokenType = 'DPoP',
        public readonly int $expiresIn = 3600,
        public readonly string $scope = '',
        public readonly ?string $refreshToken = null,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            accessToken: (string) ($data['access_token'] ?? ''),
            tokenType: (string) ($data['token_type'] ?? 'DPoP'),
            expiresIn: (int) ($data['expires_in'] ?? 3600),
            scope: (string) ($data['scope'] ?? ''),
            refreshToken: $data['refresh_token'] ?? null,
        );
    }
}
